x-pack/plugins/observability_solution/observability_ai_assistant/common/types.ts

```diff
@@ -79,9 +79,10 @@ export type ConversationUpdateRequest = ConversationRequestBase & {
 
 export interface KnowledgeBaseEntry {
   '@timestamp': string;
-  id: string;
+  id: string; // unique ID
+  doc_id?: string; // human readable ID generated by the LLM and used by the LLM to lookup and update existing entries. TODO: rename `doc_id` to `lookup_id`
```

Comment:
Where `id` must be globally unique, `doc_id` is only unique per user. Multiple entries can be assigned the same `doc_id` if they are created for different users.